contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -0,0 +1,457 @@
+        internal
+        view
+        returns (address asset, uint256 amount)
+    {
+        amount = 0;
+        if (poolAsset == wstETH) {
+            asset = stETH;
+            if (poolAmount > 0) {
+                amount = IWstETH(wstETH).getStETHByWstETH(poolAmount);
+            }
+        } else if (poolAsset == sfrxETH) {
+            asset = frxETH;
+            if (poolAmount > 0) {
+                amount = IERC4626(sfrxETH).convertToAssets(poolAmount);
+            }
+        } else {
+            asset = poolAsset;
+            amount = poolAmount;
+        }
+    }
+
+    function fromPoolAsset(address poolAsset)
+        internal
+        view
+        returns (address asset)
+    {
+        if (poolAsset == wstETH) {
+            asset = stETH;
+        } else if (poolAsset == sfrxETH) {
+            asset = frxETH;
+        } else {
+            asset = poolAsset;
+        }
+    }
+
+    /**
+     * @notice Sets max withdrawal slippage that is considered when removing
+     * liquidity from Balancer pools.
+     * @param _maxWithdrawalSlippage Max withdrawal slippage denominated in
+     *        wad (number with 18 decimals): 1e18 == 100%, 1e16 == 1%
+     *
+     * IMPORTANT Minimum maxWithdrawalSlippage should actually be 0.1% (1e15)
+     * for production usage. Contract allows as low value as 0% for confirming
+     * correct behavior in test suite.
+     */
+    function setMaxWithdrawalSlippage(uint256 _maxWithdrawalSlippage)
+        external
+        onlyVaultOrGovernorOrStrategist
+    {
+        require(
+            _maxWithdrawalSlippage <= 1e18,
+            "Max withdrawal slippage needs to be between 0% - 100%"
+        );
+        emit MaxWithdrawalSlippageUpdated(
+            maxWithdrawalSlippage,
+            _maxWithdrawalSlippage
+        );
+        maxWithdrawalSlippage = _maxWithdrawalSlippage;
+    }
+
+    /**
+     * @notice Sets max deposit slippage that is considered when adding
+     * liquidity to Balancer pools.
+     * @param _maxDepositSlippage Max deposit slippage denominated in
+     *        wad (number with 18 decimals): 1e18 == 100%, 1e16 == 1%
+     *
+     * IMPORTANT Minimum maxDepositSlippage should actually be 0.1% (1e15)
+     * for production usage. Contract allows as low value as 0% for confirming
+     * correct behavior in test suite.
+     */
+    function setMaxDepositSlippage(uint256 _maxDepositSlippage)
+        external
+        onlyVaultOrGovernorOrStrategist
+    {
+        require(
+            _maxDepositSlippage <= 1e18,
+            "Max deposit slippage needs to be between 0% - 100%"
+        );
+        emit MaxDepositSlippageUpdated(maxDepositSlippage, _maxDepositSlippage);
+        maxDepositSlippage = _maxDepositSlippage;
+    }
+
+    function _approveBase() internal virtual {
+        IERC20 pToken = IERC20(platformAddress);
+        // Balancer vault for BPT token (required for removing liquidity)
+        pToken.safeApprove(address(balancerVault), 0);
```

Comment:
I think we can just use "approve" and skip the double safeApprove()'s. The requirement for the 0 set before a change is actually local code in the local function `safeApprove`.

This will apply throughout the balancer strategy.

Reply:
Good point 👍